### PR TITLE
RUST-926 Allow direct connections to hidden nodes

### PR DESCRIPTION
**Pull request by patrickfreed**
RUST-926

This PR fixes a bug where the driver would always time out trying to determine if sessions were supported when directly connected to a hidden node.

### CODE REVIEW

File: src/client/executor.rs [hunk 1]
```diff
@@ -583,8 +584,11 @@ impl Client {
         // sessions are supported or not.
         match initial_status {
             SessionSupportStatus::Undetermined => {
+                let topology_type = self.inner.topology.topology_type().await;
                 let criteria = SelectionCriteria::Predicate(Arc::new(move |server_info| {
-                    server_info.server_type().is_data_bearing()
+                    let server_type = server_info.server_type();
+                    (matches!(topology_type, TopologyType::Single) && server_type.is_available())
```
Comment (patrickfreed) [inline src/client/executor.rs:590]: for direct connections and standalones, just try to select any server. for all other topologies, wait for a data bearing server.

File: src/sdam/description/topology/mod.rs [hunk 2]
```diff
@@ -272,6 +272,11 @@ impl TopologyDescription {
     /// Updates the topology's logical session timeout value based on the server's value for it.
     fn update_session_support_status(&mut self, server_description: &ServerDescription) {
         if !server_description.server_type.is_data_bearing() {
+            if let TopologyType::Single = self.topology_type {
+                self.session_support_status = SessionSupportStatus::Unsupported {
```
Comment (patrickfreed) [inline src/sdam/description/topology/mod.rs:276]: if the only server in the topology isn't data bearing, just mark it as not supporting sessions.

**Comment by daleione** [comment]
The document of  `src/client/options/mod.rs#L373` is still direct=true.
It's actually `directConnection`.
Please remember to fix it.

**Comment by patrickfreed** [comment]
@daleione fixed, thanks for the reminder!

**Review by abr-egn — APPROVED** [review]

File: src/client/executor.rs (hunk 1 above)
Comment (isabelatkinson) [reply src/client/executor.rs:590]: We use this same logic in `transaction_support_status` directly below, so if a user is trying to execute a transaction on a direct connection to a hidden node I believe they'll encounter the same problem (although this is a much more niche use case). Can we make these changes for `TransactionSupportStatus` as well?

Comment (patrickfreed) [reply src/client/executor.rs]: Good catch, fixed.

**Review by isabelatkinson — APPROVED** [review]

**Comment by daleione** [comment]
with this commit, query with hide node is also failed.
New error message is 
`Error: Error { kind: Command(CommandError { code: 13, code_name: "Unauthorized", message: "command find requires authentication" }), labels: {} }`.

The db link work well with mongosh. Please check, if there something missing to fix.

**Pull request merged**